### ISSUE TITLE
Profile plugin in blank when redirected with juypter_server_proxy

### ISSUE THREAD

**Issue by rsarm** [tensorflow/tensorboard#3748]
Hi. I use tensorboard within a jupyter notebook that runs remotely and it's accessible through juypterhub. Since on the remote host, tensorboard will be running on `localhost:6006`, I use `juypter_server_proxy` to redirect it through the notebook's server. I do that through monkey-patching `tensorboard.notebook._display_ipython` as follows
```python
import tensorboard
from IPython import get_ipython


def _display_ipython(port, height, display_handle):
    import os
    import IPython.display
    iframe = IPython.display.IFrame(
        src="https://<jupyter-url>/hub/user-redirect/proxy/%d/#scalars" % (os.environ['USER'], port),
        height=height,
        width="100%",
    )
    if display_handle:
        display_handle.update(iframe)
    else:
        IPython.display.display(iframe)


ipy = get_ipython()
tensorboard.notebook._load_ipython_extension(ipy)

tensorboard.notebook._display_ipython = _display_ipython
```

That works fine. The scalars, histograms, etc, show up fine, but the profile tab shows a blank page. Both within the notebook or directly on the browser. Using ssh tunnels from the host to my laptop, I checked that the profiler shows up fine. The problem seems to be with the redirecting.

I see that the blank page has an iframe `<iframe id="dashboard" src="http://<jupyter-url>/hub/user-redirect/proxy/6006/data/plugin_entry.html?name=profile"></iframe>`. I tried pasting that url directly to the browser, and it does some redirects which ends up in a blank page `http://<jupyter-url>/hub/user-redirect/proxy/6006/data/profile/index.html`.  Doing the same, with a tensorboard running un my laptop, these last redirects end up on a working profiler with a longer url `http://localhost:6006/data/plugin/profile/overview_page@;ea=20200615-211507%2Ftrain%2F2020_06_15_21_15_10;tag=overview_page@;host=nid06882`. I think with the way I'm doing the proxying through jupyter, I'm messing up with tensorboard's redirects.

Do you have some workaround for the case of JuypterHub? Otherwise could you please, point me to where I could add a fix for my use case.

**Comment by psybuzz** [comment]
Thank you for the report.  I haven't gotten around to investigation of this yet.  Could you please share your version #s for tensorboard and jupyter hub?

**Comment by rsarm** [comment]
Thanks a lot for looking into this! We have tensorboard-2.2.1 and juypterhub-1.0.0.

**Comment by rsarm** [comment]
I just tested again with `tensorboard-plugin-profile-2.3.0` and `tensorboard-2.3.0` and the profile shows up fine. Thanks!

**Comment by psybuzz** [comment]
Thanks for verifying that it works in 2.3.0.  I wasn't able to deeply investigate the problem, but since it's solved with a version bump, I suspect it might be due to the fix that landed in 2.3 - https://github.com/tensorflow/tensorboard/pull/3674.

Closing this issue.
